substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -484,6 +486,8 @@ public Path getBuilderInspectServerPath() {
             if (useJVMCINativeLibrary) {
                 builderJavaArgs.add("-XX:+UseJVMCINativeLibrary");
             } else {
+                // some JVMs need this unlocking
+                builderJavaArgs.add("-XX:+UnlockExperimentalVMOptions");
```

Comment:
Thanks for fixing this!

Reply:
@jerboaa the flag is actually present but not in the right place. With this change we essentially pass the option twice:

```Executing [
/home/zakkak/code/graal/sdk/mxbuild/linux-amd64/GRAALVM_ADB1ABFC21_JAVA11/graalvm-adb1abfc21-java11-21.3.0-dev/bin/java \
-XX:+UnlockExperimentalVMOptions \
-XX:-UseJVMCICompiler \
-XX:+UseParallelGC \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
...
```

I suggest applying the following instead:
```diff
diff --git a/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java b/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java
index 7d08509420c..8ef7d3e22dc 100644
--- a/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java
+++ b/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java
@@ -481,12 +481,6 @@ public class NativeImage {
                 }
             }
 
-            if (useJVMCINativeLibrary) {
-                builderJavaArgs.add("-XX:+UseJVMCINativeLibrary");
-            } else {
-                builderJavaArgs.add("-XX:-UseJVMCICompiler");
-            }
-
             String javaVersion = String.valueOf(JavaVersionUtil.JAVA_SPEC);
             String[] flagsForVersion = graalCompilerFlags.get(javaVersion);
             if (flagsForVersion == null) {
@@ -516,6 +510,12 @@ public class NativeImage {
                 }
             }
 
+            if (useJVMCINativeLibrary) {
+                builderJavaArgs.add("-XX:+UseJVMCINativeLibrary");
+            } else {
+                builderJavaArgs.add("-XX:-UseJVMCICompiler");
+            }
+
             return builderJavaArgs;
         }
```

Reply:
This would be a partial revert of https://github.com/oracle/graal/commit/cc11ae1d1fad550424b1fdc0a6cf7e9d6815874a. Not sure why the order was changed. @olpaw thoughts on that?

Reply:
Happy to go either way, but I thought the order changes was intentional.

Reply:
There was some reordering required to allow us to switch to running the imagbuilder on module-path when module-related options are passed to the driver. But this particular change should not cause problems as the value of useJVMCINativeLibrary is not dependent on module-related options. Thus @zakkak suggestion should be followed here.

Reply:
OK, fixed.